packages/techdocs-common/src/stages/publish/awsS3.ts

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import path from 'path';
+import express from 'express';
+import aws from 'aws-sdk';
+import { Logger } from 'winston';
+import { Entity, EntityName } from '@backstage/catalog-model';
+import { Config } from '@backstage/config';
+import { getHeadersForFileExtension, getFileTreeRecursively } from './helpers';
+import { PublisherBase, PublishRequest } from './types';
+import { ManagedUpload } from 'aws-sdk/clients/s3';
+import fs from 'fs';
+
+export class AwsS3Publish implements PublisherBase {
+  static fromConfig(config: Config, logger: Logger): PublisherBase {
+    let region = null;
+    let accessKeyId = null;
+    let secretAccessKey = null;
+    let bucketName = '';
+    try {
+      accessKeyId = config.getString(
```

Comment:
Possible improvement for a future iteration (happy to contribute if we reach there first): this should use either EC2 instance profiles/Kubernetes IRSA to remove the need to provide an access key. These credentials are automatically rotated and built into the AWS SDK.

Reply:
Maybe have the `credentials` as optional and if not present use the default credentials chain that @lowjoel mentions? Just `new aws.S3()` basically. 

We would also be happy to contribute this since we have a policy to never create users and use access keys. I can understand that this might be necessary if you are not running in AWS but still want to use S3 though.

Reply:
I'll be happy to add this to the existing PR !

@OrkoHunter Do we keep this PR in current state and iterate over it once merged to add optional credentials ? Or do we add that on this branch ?

Reply:
@ayshiff IMO let us iterate in the future! We shouldn't delay this PR more. Hopefully we can merge this next week. :)